### PR TITLE
refactor: Remove `Outcome` from network requests

### LINKED ISSUE #4909: Remove `Outcome` from network callbacks
It's never used.

### PR DESCRIPTION
**Pull request by Nerixyz**
# Description

As explained in the issue, the return value was never used.

Closes #4909.

### CODE REVIEW

File: src/providers/bttv/BttvEmotes.cpp [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include "common/NetworkRequest.hpp"
 #include "common/NetworkResult.hpp"
+#include "common/Outcome.hpp"
```
Comment (pajlada) [inline src/providers/bttv/BttvEmotes.cpp:5]: remove include

**Review by pajlada — COMMENTED** [review]
can also just remove the outcome.hpp file & its cmakelists entry
EDIT: nvm it's still used in places

**Comment by fourtf** [comment]
RIP Outcome, it was my favorite

File: src/providers/bttv/BttvEmotes.cpp (hunk 1 above)
Comment (pajlada) [reply src/providers/bttv/BttvEmotes.cpp:5]: nvm, it's actually used here

**Review by pajlada — APPROVED** [review]

**Pull request merged**